magento/magento2
# Github #31984: fix for array offset bug during some 404s (PHP 7.4)

### Description (*)


Fixes "Trying to access array offset on value of type bool" when accessing certain non existing category paths for known products, on PHP 7.4.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#31984

### Manual testing scenarios (*)

1. On PHP 7.4, as an admin, set "Generate "category/product" URL Rewrites" to "No"
2. Make a GET request to {nonexistentcategory}/{product_url} (where product_url is an existing saleable product)
3. Ensure status code is 404, not 500

### Questions or comments


### Contribution checklist (*)
